### ISSUE TITLE
IPv6 literal Host header results in invalid URLs

### ISSUE THREAD

**Issue by silverwind** [watson/original-url#3]
````js
> require("original-url")({headers: {host: '[::1]'}})
{ raw: undefined,
  protocol: 'http:',
  hostname: '::1',
  pathname: '/',
  full: 'http://::1/' }
> new URL('http://::1/')
TypeError [ERR_INVALID_URL]: Invalid URL: http://::1/
    at onParseError (internal/url.js:237:17)
    at parse (internal/url.js:246:3)
    at new URL (internal/url.js:321:5)
````

According to [rfc3986](https://tools.ietf.org/html/rfc3986#section-3.2.2), IPv6 literals must be enclosed in `[]`, but it seems `original-url` strips these brackets, forming a invalid URL.

**Comment by silverwind** [comment]
Checking it further, it appears to be a bug in the legacy `url` module of Node:

````js
> require("url").parse("http://[::1]").hostname
'::1'
> new URL("http://[::1]").hostname
'[::1]'
````
I think the proper fix would be a rewrite to use the standard [URL](https://nodejs.org/api/url.html#url_class_url) module (or [whatwg-url](https://github.com/jsdom/whatwg-url) to support node 6), but I have a workaround you might want to consider.

**Comment by watson** [comment]
The fix is released in version 1.2.2. Thanks a lot for the help 😃
